### PR TITLE
Add new contrib script for generating yaml from helm

### PR DESCRIPTION
**Pull request by LucasRoesler**
## Description

- Add new `create-static-manifest` script that will generate the static
yaml manifest files from the helm chart by using `helm template`
- Run this script during the `make charts` target, to ensure that we
update the yaml when we publish a new helm chart version
- Apply the script to the current `yaml` folder so that we can review
what, if any, changes the helm generation creates

_TODO_
[] Create values files for the arm64 and armhf manifests

## Motivation and Context

- [x] I have raised an issue to propose this change ([required](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md))

See #423 

## How Has This Been Tested?

- Basic deployment using `kubectl apply -f ./yaml`
- Waited for every component to start
- Deployed a function

_TODO_
[] validate scaling

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [x] I have signed-off my commits with `git commit -s`
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: yaml/alertmanager-dep.yml [hunk 1]
```diff
@@ -1,33 +1,54 @@
...
+            - wget
+            - --quiet
+            - --tries=1
+            - --timeout=30
+            - --spider
+            - http://localhost:9093/-/ready
+          timeoutSeconds: 30
         ports:
-        - containerPort: 9003
+        - containerPort: 9093
```
Comment (alexellis) [inline yaml/alertmanager-dep.yml]: @LucasRoesler can you add a requests value back in?

File: yaml/alertmanager-svc.yml [hunk 2]
```diff
@@ -1,15 +1,17 @@
...
   labels:
-    app: alertmanager
+    app: openfaas
+    component: alertmanager
+  name: alertmanager
+  namespace: "openfaas"
 spec:
   type: ClusterIP
   ports:
     - port: 9093
```
Comment (alexellis) [inline yaml/alertmanager-svc.yml]: Do we want targetPort? @stefanprodan

File: yaml/prometheus-svc.yml [hunk 3]
```diff
@@ -1,16 +1,17 @@
...
+    component: prometheus
+  name: prometheus
+  namespace: "openfaas"
 spec:
-  type: NodePort
+  type: ClusterIP
   ports:
     - port: 9090
       protocol: TCP
-      targetPort: 9090
```
Comment (alexellis) [inline yaml/prometheus-svc.yml]: Please update any docs which mention a NodePort of 31119 and give the kubectl port-forward equivalent before the merge of this PR.

**Comment by alexellis** [comment]
This looks great. I did some basic testing to match yours too - with `kind`.

I left a few comments for you.

File: yaml/alertmanager-svc.yml (hunk 2 above)
Comment (stefanprodan) [reply yaml/alertmanager-svc.yml]: No, I think it is ok to remove it.

File: yaml/alertmanager-dep.yml (hunk 1 above)
Comment (LucasRoesler) [reply yaml/alertmanager-dep.yml]: 👍  The best way to do that is to add it to the helm chart

File: yaml/prometheus-svc.yml (hunk 3 above)
Comment (LucasRoesler) [reply yaml/prometheus-svc.yml]: I could not find any explicit reference to this port

Comment (LucasRoesler) [reply yaml/prometheus-svc.yml]: actually just found one in the docs, my search missed that repo.  The only reference is here, and it doesn't really say how to use it.  I could just remove the line?

Comment (LucasRoesler) [reply yaml/prometheus-svc.yml]: I added a small section to the chart readme to describe accessing prometheus, we will need a separate PR to the docs repo to replicate or remove the instructions there too

Comment (alexellis) [reply yaml/prometheus-svc.yml]: It's also in the readme potentially.

Comment (LucasRoesler) [reply yaml/prometheus-svc.yml]: I checked the readme of faas-netes and the chart and then in the docs project, it was only directly referenced in the docs project

**Comment by alexellis** [comment]
Shall I merge?

**Comment by LucasRoesler** [comment]
I was going to add values files for the arms folders too so that those can also be generated

**Comment by alexellis** [comment]
Let's do it as a second PR?

**Comment by LucasRoesler** [comment]
do you want me to squash/rebase?

**Comment by LucasRoesler** [comment]
I squashed the commits

**Pull request merged**